src/ipld.rs

```diff
@@ -363,4 +388,12 @@ mod tests {
         let ipld = Ipld::Map(map);
         assert_eq!(ipld.get("a").unwrap(), Some(&Ipld::Integer(0)));
     }
+
+    // NaN floats are forbidden in the IPLD Data Model, but still make sure they are treated as
+    // equal in case they accidentally end up there.
+    #[test]
+    fn test_partial_eq_nan() {
```

Comment:
What happens when you compare different nans? Try a 3-way comparison between `f64::NAN` with `f64::from_ne_bytes([0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0xf0, 0x7f])` and `f64::from_ne_bytes([0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x7f])`, they're all going to be "equal" aren't they? How much do we care about such things? (Also one of the reasons that NaN is forbidden).

Reply:
IMO, it's okay to define these as equal. Especially given that we'll reject all these when we serialize and deserialize them.